src/services/Elements.php

```diff
@@ -1376,6 +1378,11 @@ public function duplicateElement(
         // Clean up our tracks
         $mainClone->duplicateOf = null;
 
+        // discard provisional draft from the original element
+        if ($originalProvisionalDraftId) {
+            $this->deleteElementById($originalProvisionalDraftId);
+        }
+
```

Comment:
This doesn’t feel worth the extra argument on `duplicateElement()`. Could have just as easily been a check within `ElementsController::actionDuplicate()` after the duplication is successful.

```php
if ($element->isProvisionalDraft) {
    Craft::$app->getElements()->deleteElement($element);
}
```

Reply:
agreed; I've pushed an update